### DISCUSSION TITLE
How to reference env vars from rule files

### DISCUSSION THREAD

**Discussion by missaeldenadai** [jertel/elastalert2#1340] [Q&A] [answered]
Hi,

Is there a way to reference env vars from rules files?
I need to inject the Opsgenie API key in the rules and I cannot use the `extraConfigOptions` because rules may use different API keys.
Also my rules are defined in separate files, so I cannot use `envsubst` as well.

Thank you.

**Comment by jertel** [comment, marked as answer]
You can use [reference values](https://elastalert2.readthedocs.io/en/latest/alerts.html#alerter-base-type) in combination with `extraConfigOptions` to do assign different keys per rule. Ex:

Rule 1:
```
name: foo
alert: opsgenie
opsgenie_key: $fookey$
```

Rule 2:
```
name: bar
alert: opsgenie
opsgenie_key: $barkey$
```

extraConfigOptions:
```
fookey: xyz
barkey: abc
```

**Reply by missaeldenadai** [reply]
It worked like a charm!
Thanks a lot!
